### DISCUSSION TITLE
我为什么不再喜欢 Go 了

### DISCUSSION THREAD

**Discussion by giscus** [JmPotato/Pomash#14] [Announcements]
# 我为什么不再喜欢 Go 了

https://ipotato.me/article/69

**Comment by auula** [comment]
不错写着不错，Go 语法简单入门简单上手快，我曾经也是对 Go 很吹捧，写了几个项目之后感觉不如 Java ，Go 厉害之处就是通过 goroutine 和 CSP 抽象使得开发者对并行编程没有难度，其他方面我到觉得不过如此，写写服务端和中间件层，其他领域就算了。

**Reply by auula** [reply]
看了你说 AST 那段，我也遇到这种情况 [https://github.com/auula/gsql](https://github.com/auula/gsql) ,  各种断言问题，现在的范型也是一个简易版本的，不如 C# 的范型和 Java ，Java 范型虽然为了兼容老的版本搞了类型擦除，但是 GO 范型用起来确实不如它们，我博客也写过几篇这样的文章 [https://www.ibyte.me/index.php/tag/go/](https://www.ibyte.me/index.php/tag/go/) 。

**Reply by auula** [reply]
研究 PL 可以去研究 [graalvm.org](http://graalvm.org) 这个项目。

**Reply by auula** [reply]
另外我对 Rust 也不感冒，上手难度和目前应用场景在基础软件之上，并且 Rust 中文社区的一些让我产生厌恶感。
